client.go

```diff
@@ -62,7 +63,9 @@ type Client struct {
 	halfOpenLimit int
 	peerID        [20]byte
 	// The net.Addr.String part that should be common to all active listeners.
-	listenAddr     string
+	listenAddr string
+	// listen UDP addr, may be different from TCP port because of UPNP
+	listenUDPAddr  string
```

Comment:
Peers expect the TCP and UDP ports to be the same. How do you announce to DHT, trackers, and over PEX with this?